inc/Engine/Preload/Admin/Subscriber.php

```diff
@@ -14,22 +25,37 @@ class Subscriber implements Subscriber_Interface {
 	protected $settings;
 
 	/**
-	 * Creates an instance of the class.
+	 * Clear cache controller.
+	 *
+	 * @var ClearCache
+	 */
+	protected $controller;
+
+	/**
+	 * Instantiate Subscriber.
 	 *
-	 * @param Settings $settings Settings instance.
+	 * @param Options_Data $options Options instance.
+	 * @param Settings     $settings Preload settings.
+	 * @param ClearCache   $clear_cache Clear cache controller.
 	 */
-	public function __construct( Settings $settings ) {
-		$this->settings = $settings;
+	public function __construct( Options_Data $options, Settings $settings, ClearCache $clear_cache ) {
+		$this->options    = $options;
+		$this->settings   = $settings;
+		$this->controller = $clear_cache;
 	}
 
 	/**
-	 * Return an array of events that this subscriber listens to.
+	 * Return an array of events that this subscriber wants to listen to.
 	 *
 	 * @return array
 	 */
 	public static function get_subscribed_events() {
 		return [
-			'admin_notices' => [ 'maybe_display_preload_notice' ],
+			'admin_notices'                          => [ 'maybe_display_preload_notice' ],
+			'after_rocket_clean_post'                => [ 'clean_partial_cache', 10, 3 ],
+			'after_rocket_clean_term'                => [ 'clean_partial_cache', 10, 3 ],
+			'rocket_after_clean_terms'               => 'clean_urls',
+			'rocket_after_preload_after_purge_cache' => 'clean_full_cache',
```

Comment:
The full cache purge should be on the `after_rocket_clean_domain` action instead